playbooks/dataiku.yml

```diff
@@ -43,45 +41,77 @@
 
     - name: Define DATA_DIR based on attached storage
       set_fact:
-          DATA_DIR: "/data/volume_2"
+          dss_data_dirs_location: "/data/volume_2/dataiku"
       when: extra_volume_attached.stat.exists
 
-    - name: create data directory for Dataiku DSS when there is no attached storage
-      file:
-        path: /data/dataiku
-        mode: '0777'
-        state: directory
-      when: not extra_volume_attached.stat.exists
-
     - name: Define DATA_DIR based on attached storage
       set_fact:
-          DATA_DIR: "/data/dataiku"
+          dss_data_dirs_location: "/data/dataiku"
       when: not extra_volume_attached.stat.exists
 
-    - name: Debug output
-      debug:
-        msg: "{{ DATA_DIR }}"
+    - name: Create private directories
```

Comment:
You use the plural "directories", but only one is made for the dss_service_user... So anyone who wishes to use it, needs to become this user first? Have you considered looping over all users on the system? If yes, check out this role we created to gather all users on a system: https://github.com/UtrechtUniversity/researchcloud-items/blob/main/roles/fact_regular_users/tasks/main.yml